### PR TITLE
Update guide to use simplified remote load.

### PR DESCRIPTION
**Pull request by bianpengyuan**
Also update various guide to be based on Istio release 1.9.

Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

### CODE REVIEW

File: extensions/basic_auth/config/gateway-filter.yaml [hunk 1]
```diff
@@ -12,12 +12,30 @@ spec:
         filterChain:
           filter:
             name: envoy.http_connection_manager
-      proxy:
```
Comment (richardwxn) [inline extensions/basic_auth/config/gateway-filter.yaml:15]: should we keep the proxy version match for 1.9?

File: extensions/basic_auth/README.md [hunk 2]
```diff
@@ -1,59 +1,33 @@
...
-    sha256: 707e29db817f76c974d7ce1fe2f61ad64c88856c7ddba99a36fe95439bfe1281
-  ```
-
-After applying the filter, gateway should start enforce the basic auth rule. Use `productpage` app as an example, to test that the rule works, you can curl with and without the authorization header. For example
+In the following guide we will configure Istio proxy to download and apply Basic Auth filter.
+
+Two `EnvoyFilter` resources will be applied, to inject basic auth filter into HTTP filter chain.
+For example, [this configuration](./config/gateway-filter.yaml) injects the basic auth filter to `gateway`.
+
+The first `EnvoyFilter` will be inject an HTTP filter into gateway proxies.
```
Comment (richardwxn) [inline extensions/basic_auth/README.md:19]: nit: will be inject/will inject

File: extensions/basic_auth/README.md [hunk 3]
```diff
@@ -1,59 +1,33 @@
...
+It is configured to request the extension configuration named as `istio.basic_auth` from `ads` (i.e. Aggregated Discovery Service), which is the same configuration source that Istiod uses to provide all other configuration resources.
+Along with the configuration source, the initial fetch timeout is also set, in order to prevent filters with slow or failed Wasm module remote fetch becomes effective.
+The second `EnvoyFilter` resource provides configuration for the filter, which is composed as an `EXTENSION_CONFIG` patch and will be distributed to the proxy as an Envoy [`Extension Configuration`](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS) resource.
+Once this update reaches Istio agent, it will download the Wasm module file and store it at the local file system.
+If the download fails, Istio-agent will reject the `Extension Configuration` update and prevent bad Wasm filter configuration from reaching Envoy.
```
Comment (richardwxn) [inline extensions/basic_auth/README.md:24]: do we have a stats metric for ECDS update failure? if so can we add the name here as well so users can monitor

File: extensions/basic_auth/README.md [hunk 4]
```diff
@@ -104,10 +79,13 @@ message BasicAuth {
...
 ## Feature Request and Customization
+
 ---
 
 If you have any feature request or bug report, please open an issue in this repo. Currently it is on the roadmap to:
-- [ ] Read secret from local file. This is pending on Wasm ABI supporting file read.
-- [ ] Add regex to path matching
 
-It is recommended to customize the extension according to your needs. Please take a look at [Wasm extension C++ development guide](../doc/write-a-wasm-extension-with-cpp.md) for more information about how to write your own extension.
+* [ ] Read secret from local file. This is pending on Wasm ABI supporting file read.
```
Comment (richardwxn) [inline extensions/basic_auth/README.md:87]: Do we have an opening issue in WASM ABI repo?

File: extensions/basic_auth/config/gateway-filter.yaml (hunk 1 above)
Comment (bianpengyuan) [reply extensions/basic_auth/config/gateway-filter.yaml:15]: Done.

File: extensions/basic_auth/README.md (hunk 3 above)
Comment (bianpengyuan) [reply extensions/basic_auth/README.md:24]: Yeah it is already documented on istio.io, and there is actually lots of duplication here to that document: https://istio.io/latest/docs/ops/configuration/extensibility/wasm-module-distribution/

I removed bunch of content from the readme guide and redirect user to read the official istio.io guide instead.

File: extensions/basic_auth/README.md (hunk 4 above)
Comment (bianpengyuan) [reply extensions/basic_auth/README.md:91]: Created one and linked here.

**Review by richardwxn — APPROVED** [review]

**Pull request merged**